src/js/ctrl/export.ts

```diff
@@ -37,8 +37,12 @@ let counts = duplicate(SPEC_COUNT);
  * direct-manipulation interactors (handles, connectors, etc.).
  * @returns {Object} A Vega specification.
  */
-export function exporter(internal: boolean = false): Spec {
-  const state = store.getState();
+export function exporter(internal: boolean = false, history): Spec { // TODO: use history.ts so you only have to pass id or index
+  if (history && history.getIn(['marks']).size == 0) {
+    return {};
+  }
+
+  const state = history || store.getState().getIn(['vis', 'present']);
```

Comment:
i would change the parameter from `history` to `visState?: VisStateTree` and do `visState = visState || store.getState().getIn(['vis', 'present']);`

this makes it clearer what the export function does.